modules/plugins/tabline/nvim-bufferline/config.nix

```diff
@@ -9,9 +9,7 @@
   inherit (lib.nvim.lua) toLuaObject;
 
   cfg = config.vim.tabline.nvimBufferline;
-
-  self = import ./nvim-bufferline.nix {inherit lib;};
-  inherit (self.options.vim.tabline.nvimBufferline) mappings;
+  inherit (cfg) mappings;
```

Comment:
The reason your system configuration is failing to build is because you cannot get mapping description from `config`. It holds the type defined in `mkOption` and not and the `option` type itself. You need to import the module to get it.

Reply:
Ah so that's what self was doing there, thanks.